pyproject.toml

```
@@ -1,6 +1,268 @@
+
+# Include license file in base/root of repository
+[license]
+file = "LICENSE"
+
+[tool.distutils.bdist_wheel]
+universal = true
+
+[tool.hatch.version]
+path = "elyra/_version.py"
+
+[tool.hatch.build]
+ignore-vcs = true
+
+# Include data files with dist
+[tool.hatch.build.targets.wheel.shared-data]
+"etc/config/jupyter_notebook_config.d" = "etc/jupyter/jupyter_notebook_config.d"
+"etc/config/jupyter_server_config.d" = "etc/jupyter/jupyter_server_config.d"
+"etc/config/settings" = "etc/jupyter/labconfig"
+"etc/config/metadata/runtime-images" = "share/jupyter/metadata/runtime-images"
+"etc/config/metadata/component-catalogs" = "share/jupyter/metadata/component-catalogs"
+"etc/config/components" = "share/jupyter/components"
+"etc/config/components/kfp" = "share/jupyter/components/kfp/"
+"etc/config/components/airflow" = "share/jupyter/components/airflow/"
+"build/labextensions" = "share/jupyter/labextensions"
+
+# Specify which files to include and exclude in source distribution
+[tool.hatch.build.targets.sdist]
+include = [
+    "build/labextensions",
+    "elyra",
+    "etc/config/components/*.json",
+    "etc/config/components/kfp*/.yaml",
+    "etc/config/jupyter_notebook_config.d/*.json",
+    "etc/config/jupyter_server_config.d/*.json",
+    "etc/config/metadata/runtime-images/*.json",
+    "etc/config/metadata/component-registries/*.json",
+    "etc/config/settings/*.json",
+]
+exclude = [
+    "*.ipynb",
+    "*.py[cod]",
+    "__pycache__",
+    ".git",
+    ".gitignore",
+    ".ipynb_checkpoints",
+    ".DS_Store",
+    "*.sh",
+    "docs",
+]
+
+# Move flake8 configuration from setup.cfg. Use pflake8 instead of flake8.
+# pflake8 is a flake8 wrapper compatible with pyproject.toml
+[tool.flake8]
+application-import-names = ["elyra", "test"]
+application-package-names = ["elyra", "test"]
+enable-extensions = "G"
+max-line-length = 120
+# References:
+# https://flake8.readthedocs.io/en/latest/user/configuration.html
+# https://flake8.readthedocs.io/en/latest/user/error-codes.html
+ignore = [
+    # Import formatting
+    "E4",
+    # Comparing types instead of isinstance
+    "E721",
+    # Assigning lambda expression
+    "E731",
+    # Ambiguous variable names
+    "E741",
+    # File contains nothing but comments
+    "H104",
+    # Include name with TODOs as in # TODO(yourname)
+    "H101",
+    # Enable mocking
+    "H216",
+    # Multi line docstrings should start without a leading new line
+    "H404",
+    # Multi line docstrings should start with a one line summary followed by an empty line
+    "H405",
+    # Line break before binary operator
+    "W503",
+    # Allow breaks after binary operators
+    "W504"
+]
+import-order-style = "google"
+
+# Configuration parameters for the black code formatting tool
 [tool.black]
 # max line length; same as 'max-line-length' in setup.cfg
 line-length = 120
 
 # Python versions that should be supported by the produced output
 target-version = ['py37', 'py38', 'py39', 'py310']
```

Although not added in #3033, we should include a 3.11 reference...
```suggestion
target-version = ['py37', 'py38', 'py39', 'py310', 'py311']
```